File: C++/2015/Pie Day.cpp
#include <iostream>

using namespace std;

int ans = 0;

void pi(int pieces, int people, int min) {
    if (people == 1) {
        ans++;
        return;
    }

    for (size_t i = min; i < ((pieces / people) + 1); i++)
    {
        pi(pieces - i, people - 1, i);
    }
    
}

int main() {
    int n, k;
    
    cin >> n;
    cin >> k;

    pi(n, k, 1);
    
    cout << ans << endl;
}

File: Python/2015/Pie Day.py
n = int(input())
k = int(input())
ans = 0 

def pi(pieces, people, minimum):
    global ans
    if people == 1:
        ans += 1
        return
    for i in range(minimum, pieces//people + 1):
        pi(pieces - i, people - 1, i)

pi(n, k, 1)
print(ans)
